actors/builtin/miner/miner_state.go

```diff
@@ -236,6 +236,18 @@ func (st *State) ComputeProvingSet(store adt.Store) ([]abi.SectorInfo, error) {
 	return sectorInfos, nil
 }
 
+func (st *State) ComputeWindowedPoStChallengeCount(store adt.Store) (uint64, error) {
+	provingSet, err := st.ComputeProvingSet(store)
```

Comment:
This is very slow. I think we can do something faster but it will depend on changes and discipline around the fault representation so that we can safely just subtract the fault count from the sector count. Please leave a comment here: `TODO: this is too slow, subtract fault count from sector count when these are maintained consistently`.

Reply:
done!